optuna/_study_summary.py

```diff
@@ -77,3 +91,33 @@ def __le__(self, other: Any) -> bool:
             return NotImplemented
 
         return self._study_id <= other._study_id
+
+    @property
+    def direction(self) -> StudyDirection:
+
+        if len(self._directions) > 1:
+            raise RuntimeError(
+                "This attribute is not available during multi-objective optimization."
+            )
+
+        return self._directions[0]
+
+    @direction.setter
+    def direction(self, d: StudyDirection) -> None:
+
+        if len(self._directions) > 1:
+            raise RuntimeError(
+                "This attribute is not available during multi-objective optimization."
+            )
+
+        self._directions = [d]
+
+    @property
+    def directions(self) -> Sequence[StudyDirection]:
+
+        return self._directions
+
+    @directions.setter
+    def directions(self, d: Sequence[StudyDirection]) -> None:
+
+        self._directions = list(d)
```

Comment:
ditto.

Reply:
The setter method of `StudySummary.directions` is used in the Redis storage [here](https://github.com/optuna/optuna/pull/1994/files#diff-f050d3fb6c53d21907e341411c630bd5e31bf175a650494dfdfeb576ae73080eL189-R193). This is the only use of the setter. Do you think we should create a new instance of `StudySummary` and remove the setter?

Reply:
> Do you think we should create a new instance of `StudySummary` and remove the setter?

No, I don't think so. Alternatively, how about accessing `StudySummary._directions` there? The attributes prefixed with "_" are accessible inside the package (c.f., https://github.com/optuna/optuna/wiki/Coding-Style-Conventions#add-_-prefix-to-the-names-of-private-methods-functions-fields-and-classes).

```python
study_summary._directions = list(directions)
```

Reply:
Thanks for the detailed explanation. I have followed your suggestion.